Написал Query builder для формирования sql запроса

Sonicxsz opened the pull request:
Подогнал user update и catalog update под Query builder ----------------------------------------------------- Добавил id в jwt токен для легкой идентификации запроса юзера Добавил ручку для update юзера
----------------------------------------------------- Добавил задержку при проверке пароля в логине для избежания time attack

Sonicxsz committed:
Написал Query builder для формирования sql запроса
Подогнал user update и catalog update под Query builder
-----------------------------------------------------
Добавил id в jwt токен для легкой идентификации запроса юзера
Добавил ручку для update юзера
-----------------------------------------------------
Добавил задержку при проверке пароля в логине для избежания time attack

Holodeec commented on configs/server.toml:
```diff
@@ -3,8 +3,8 @@ log_level="warn"
 [storage]
-db_conn_string="postgres://user:user@postgres:5432/postgres?sslmode=disable"
-db_migrations_url="postgres://user:user@postgres:5432/postgres?sslmode=disable"
+db_conn_string="postgres://user:user@localhost:5432/postgres?sslmode=disable"
```
что это?

Sonicxsz replied:
Надо отдельный конфиг сделать для запуска приложения в режиме разработки, а так:
Если запустить приложение с помощью докера то можно напрямую по имени контейнера делать запрос, а если локально то надо по localhost стучаться

Holodeec commented on internal/repository/user_repository.go:
```diff
@@ -17,31 +18,44 @@ func NewUserRepository(pool *pgxpool.Pool) *UserRepository {
 
-func (ur *UserRepository) Create(cxt context.Context, u *model.User) (*model.User, error) {
-	err := ur.db.QueryRow(cxt, insertUser, u.Email, u.Username, u.Password).Scan(&u.Id)
+func (ur *UserRepository) Create(cxt context.Context, u *dto.UserCreateRequest) error {
```
в репо мы должны работать с сущностью а не с дто

Holodeec commented on internal/service/user_service.go:
```diff
@@ -74,24 +91,73 @@ func (s *UserService) GetUser(ctx context.Context, email string) (*dto.UserGetRe
+
+	query, values := qb.BuildUpdateQuery("public.users", "id", req.Id)
+
+	// Юзер не предоставил данные для измения(пустой запрос)
```
это должно быть в валидации

Sonicxsz replied:
Перенес в валидатор

Holodeec commented on internal/service/user_service.go:
```diff
@@ -74,24 +91,73 @@ func (s *UserService) GetUser(ctx context.Context, email string) (*dto.UserGetRe
+}
+
+func (s *UserService) UpdateUserAddress(cxt context.Context, req *dto.UserAddressUpdateRequest) error {
+	// Подумать нужно ли перенести формирование sql запроса на уровень репозитория
```
нужно

Sonicxsz replied:
по хз как это сделать без создания однотипных методов в репозитории, оставим на подумать

Sonicxsz committed:
Поправил тип в user repository

Sonicxsz committed:
Удалил ненужные модели
Убрал валидацию данных из сервиса user и перенес в IsValid
Изменил структуру валидатора, валидатор теперь собирает количество полей которые были провалидированы
Добавил метод AddError который позволяет из внешнего мира добавить ошибку в срез ошибок

Sonicxsz committed:
Тестовый
